doc/memos/rdm-draft-buschsieweke-timer-api.md

```diff
@@ -0,0 +1,201 @@
+4. The Etimer library posts events on timer expiration and a thread can block
+   until a specific event has been posted. Combined, this can be used to
+   delay execution
+5. Only available when using TOSTThreads
+
+# 2. Requirements
+
+The following lists of requirements consist of qualitative and quantitative
+requirements. For the quantitative requirements, any improvement will enable
+more scenarios to use the timer API. Thus, no meaningful lower bounds can be
+specified. Instead, a set of benchmarks will be developed that allow to
+continuously improve the performance and to detect any possible regressions of
+future changes.
+
+## Feature Requirements
+
+- Software Timers
+    - Upper bound for number of timers only given by available resources
+    - APIs to set and abort timers
+    - Support for periodic timers (support via explicit calls to some
+      `timer_reset()` like in Contiki is fine)
+    - Long term low resolution software timers that play well with power management
+    - Short term high resolution software timers
+- Delay of program execution
+    - API to block calling thread for long periods and allows other threads to run
+    - Long term delays must play well with power management
+    - Highly precise short term delays (e.g. for bit banging)
+- System time
+    - APIs to get the current system time in reasonable units
+- Utilities for convenience
+    - Timer utilities
+        - Timer that send a message upon expiration
+        - Timer that post an event upon expiration (to allow execution in thread context)
+        - Timer that wakes up a thread upon expiration
+    - Timeout utilities
+        - Timeout for receiving a message
+        - Timeout for locking a mutex
+
+## Real-Time Requirements
+
+- The implementation must only disable IRQs for short periods of time
+- The implementation must cause as little IRQs as possible and its ISRs must
+  be as short as possible
+- The implementation should provide means to execute callbacks in thread
+  context
+
+## Precision Requirements
+
+- Timer callbacks should never fire prematurely (+- 1 tick of the underlying
+  timer)
+- The accuracy of execution delay should be measured with single digit micro
+  seconds for short delays
+- Timer callbacks should fire as early as possible after the timer expired
+
+## Performance Requirements
+
+- Efficient use of RAM/ROM
+- Optimized for power management
+    - The timer API should only cause the system to wake up when strictly needed
+- Efficient use of CPU cycles
+
+# 3. Design Decisions
+
+TODO: Add these after some discussion.
+
+## Acknowledgements
+
+Thanks to K. Schleiser, J. Nohlgård, M. Rottleuthner, and all other contributors
+to this document and RIOT's timer API.
+
+TODO: Extend as needed.
+
+## References
+
+- [RFC7228 - Terminology for Constrained-Node Networks](https://tools.ietf.org/html/rfc7228)
+- TODO: Extend as needed
+
+## Revision
+
+TODO: Add revisions, if needed
+
+## Contact
+
+TODO: Do we need this?
```

Comment:
@maribu Yes we do! The goal is to give a handle to people who might have about questions about the doc later down the line...
(similar to an IETF draft or RFC lists author's email/postal address at the end of the doc).

Reply:
How about the devel mailing list to reduce the bus factor here?

Reply:
What mentioning both 
- main author(s) contact(s) and 
- a default: for that would maintainers@ be better than devel@ ?

Let's try to both reduce the bus factor *and* to increase named responsiveness in most cases ;)